data/avro-schema/src/main/resources/avro/net/corda/data/flow/event/session/SessionInit.avsc

```diff
@@ -26,6 +26,22 @@
       "type": ["null","string"],
       "doc": "Flow ID of the flow initiating the session. This is only required to inform the Flow Mapper. It should be null when sent to counterparties."
     },
+    {
+      "name": "contextUserProperties",
+      "type": {
+        "type": "map",
+        "values": "string"
+       },
```

Comment:
We might need to store these as arrays rather than maps, because I don't think the serializer is guaranteed to serialize the entries in the same order every time, which can result in problems when signing/verifying serialized objects.